starlite/utils/serialization.py

```diff
@@ -31,22 +34,36 @@
     # pydantic specific types
     BaseModel: lambda m: m.dict(),
     ByteSize: lambda b: b.real,
-    EmailStr: str,
     NameEmail: str,
     Color: str,
-    AnyUrl: str,
     SecretField: str,
-    ConstrainedInt: int,
-    ConstrainedFloat: float,
-    ConstrainedStr: str,
     ConstrainedBytes: lambda b: b.decode("utf-8"),
-    ConstrainedList: list,
-    ConstrainedSet: set,
-    ConstrainedFrozenSet: frozenset,
     ConstrainedDecimal: float,
     ConstrainedDate: lambda d: d.isoformat(),
-    PaymentCardNumber: str,
-    StrictBool: int,  # pydantic compatibility
+    IPv4Address: str,
+    IPv4Interface: str,
+    IPv4Network: str,
+    IPv6Address: str,
+    IPv6Interface: str,
+    IPv6Network: str,
+    # pydantic compatibility
+    deque: list,
+    Decimal: decimal_encoder,
+    StrictBool: int,
+    Pattern: lambda o: o.pattern,
+    # support subclasses of stdlib types, e.g. pydantic's constrained types. If no
+    # previous type matched, these will be the last type in the mro, so we use this to
+    # (attempt to) convert a subclass into its base class.
+    # see https://github.com/jcrist/msgspec/issues/248
+    # and https://github.com/starlite-api/starlite/issues/1003
+    str: str,
+    int: int,
+    float: float,
+    list: list,
+    tuple: tuple,
+    set: set,
+    frozenset: frozenset,
+    dict: dict,
```

Comment:
Since subclasses of `dict`, `list`, and `tuple` are handled natively, those keys will never be hit here and can be removed. I'm also happy to add native support for subclasses of `set` and `frozenset` for the same reasons I gave before (but for now we won't support `str`/`int`/`float` subclasses).

Reply:
This is just to support the edge case of using instances of constrained classes. See my comment below.

Native support for subclassed `set` and `frozenset` would be very much welcome though!

Reply:
I understand that, I'm just saying that because `dict`, `list`, and `tuple` (and in the next release `set`/`frozenset`) are natively supported, those entries in this mapping aren't necessary because a subclass of these will never result in `enc_hook` being called.

Reply:
You know what, you're right. I previously thought that there's some edge case where this was needed, but I can't reproduce it anymore, so I guess it was a fluke on my side. I'll remove this. Thanks for the feedback here!